packages/modules/providers/payment-stripe/src/core/stripe-base.ts

```diff
@@ -321,21 +389,23 @@ abstract class StripeBase extends AbstractPaymentProvider<StripeOptions> {
     }))
   }
 
-  async savePaymentMethod(
-    input: SavePaymentMethod
-  ): Promise<PaymentProviderError | SavePaymentMethodResponse> {
-    const { context, data } = input
-    const customer = context?.customer
+  async savePaymentMethod({
+    context,
+    data,
+  }: SavePaymentMethodInput): Promise<SavePaymentMethodOutput> {
+    const accountHolderId = context?.accountHolder?.data?.id as
+      | string
+      | undefined
 
-    if (!customer?.metadata?.stripe_id) {
-      return this.buildError(
+    if (!accountHolderId) {
+      throw this.buildError(
         "Account holder not set while saving a payment method",
         new Error("Missing account holder")
       )
     }
 
     const resp = await this.stripe_.setupIntents.create({
-      customer: customer.metadata.stripe_id as string,
+      customer: accountHolderId,
```

Comment:
question: Same as before, wouldn't this need to be the `external_id`?

Reply:
Answered above